### PR TITLE
fix: fix amplitude backend tracking

### PR DESCRIPTION
**Pull request by Sparkier**
Also only tracks when in prod.

fix EOT-44

### CODE REVIEW

File: src/lib/server/amplitude.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { dev } from '$app/environment';
+import { track } from '@amplitude/analytics-node';
+
+export function trackEvent(
+	name: string,
+	eventProperties: Record<string, unknown> | undefined,
```
Comment (cabreraalex) [inline src/lib/server/amplitude.ts:6]: you can make this optional and the last parameter, then when calling the function you can omit the parameter if we don't pass it in the calls.

File: src/routes/(app)/home/+page.server.ts [hunk 2]
```diff
@@ -57,7 +57,7 @@ export const actions = {
 				body: "Couldn't create project"
 			};
 		} else {
-			track('Prompt Created', { user_id: session.user.id });
+			trackEvent('Prompt Created', undefined, { user_id: session.user.id });
```
Comment (cabreraalex) [inline src/routes/(app)/home/+page.server.ts:60]: e.g. here you can then just do `trackEvent('Prompt Created', { user_id: session.user.id })`

**Review by cabreraalex — CHANGES_REQUESTED** [review]

File: src/lib/server/amplitude.ts (hunk 1 above)
Comment (cabreraalex) [reply src/lib/server/amplitude.ts:6]: by optional I mean doing `eventProperties?:`

Comment (Sparkier) [reply src/lib/server/amplitude.ts]: Does not work exactly like that have to swap the params but I think still better.

**Review by cabreraalex — APPROVED** [review]

**Pull request merged**